src/provider/states/create_config.rs

```diff
@@ -0,0 +1,296 @@
+                        }
+                    } else {
+                        debug!("No update needed for {:?}", target_file);
+                    }
+                } else {
+                    info!("No content found for key {}", key);
+                }
+            }
+        } else {
+            debug!("No data found in ConfigMap..");
+        }
+        Ok(())
+    }
+
+    fn needs_update(target_file: &PathBuf, content: &str) -> Result<bool, StackableError> {
+        if target_file.is_file() {
+            let current_content = read_to_string(target_file)?;
+            debug!("Compared config file {:?} with result of", target_file);
+            return Ok(current_content.ne(content));
+        }
+        debug!(
+            "Target config file {:?} doesn't exist, no need to compare.",
+            target_file
+        );
+        Ok(true)
+    }
+}
+
+#[async_trait::async_trait]
+impl State<PodState> for CreatingConfig {
+    async fn next(
+        mut self: Box<Self>,
+        pod_state: &mut PodState,
+        _pod: &Pod,
+    ) -> Transition<PodState> {
+        let name = _pod.name();
+        let client = pod_state.client.clone();
+        let package = pod_state.package.clone();
+        let config_directory = pod_state.config_directory.clone();
+        //self.target_directory = Some(config_directory.join(package.get_directory_name()));
+        //let target_directory = self.target_directory.clone().unwrap();
+        let target_directory = pod_state.get_service_config_directory();
+
+        // Check if all required config maps have been created in the api-server
+        let referenced_config_maps = self.get_config_maps(_pod).await;
+        let missing_config_maps = self
+            .missing_config_maps(client.clone(), referenced_config_maps)
+            .await;
+        if !missing_config_maps.is_empty() {
+            // not all configmaps are present
+            info!("Missing config maps, waiting..");
+            return Transition::next(
+                self,
+                WaitingConfigMap {
+                    missing_config_maps,
+                },
+            );
+        }
+
+        debug!("Entering state \"creating config\" for service {}", name);
+        let containers = _pod.containers();
+        if containers.len().ne(&1) {
+            let e = PodValidationError {
+                msg: "Only pods containing exactly one container element are supported!"
+                    .to_string(),
+            };
+            fail_fatal!(e);
+        }
+        let container = containers[0].clone();
+
+        if let Some(volumes) = _pod.volumes() {
+            debug!("Found {} volumes in pod {}", volumes.len(), _pod.name());
+            if let Some(mounts) = container.volume_mounts() {
+                debug!("Found {} mounts in pod {}", mounts.len(), _pod.name());
+                // Got mounts and volumes, we can now decide which ones we need to act upon
+                for mount in mounts {
```

Comment:
We probably could, yes.
I do think that the way that it is currently written makes it easier to grasp what is happing here at a glance though. It would be nice to safe a level of indentation, however the way it is currently makes it fairly obvious where another level of iteration / condition starts, whereas with your proposal we'd roll two loops into one line..
It is a matter of personal preference I think - and mine is not a strong one, so if you feel I should change this I'm happy to!